openlibrary/templates/covers/book_cover_work.html

```diff
@@ -10,7 +10,8 @@
 $if size == "M":
     <div class="coverMagic">
             <div class="SRPCover bookCover" style="display: $cond(cover_url, 'block', 'none');">
-                <a href="$cover_lg" class="coverLook" title="View a larger book cover"><img src="$cover_url" class="cover" alt="Cover of: $title $_('by') $author_names"/></a>
+                <a href="$cover_lg#seeImage"
```

Comment:
Ooooh, not crazy about using the anchor's href. The `href` is usually specified in these cases as a sort of fallback; if the user `ctrl-click`s or `middle-click`s, then the anchor's href gets opened in a new tab. Looking at this HTML, I would never in a million years be able to guess what it's doing. I would think that `#seeImage` was an identifier _on_ `$cover_lg`; not an element on the current page. Or, I would think the dialog would open an iframe to the `href`.

(One of the main things is trying to make code readable from multiple angles; not just the function definition, but also the invocation should be easy to understand. When looking at an invocation, one shouldn't have to look at the function's definition to understand what is going to happen. That's not quite the case here).

Reply:
I prefer to use well understood attributes over made up data attributes where I can.

Given a hash fragment points to a document fragment in all these cases it made sense to use href (and i was thinking possibly with :target selector we'd be able to make these backwards compatible. I think in the case here yes it's wrong.. as the href points to an image without a fragment identifier.

Maybe in this case the magic dialog--open class should not be used but it should be wired up manually (it's a progressive enhancement in this case not a javascript only thing).

The alternative is to use data-dialog-href attribute instead of href but that feels icky here.

What do you prefer?
My hope is to overhaul all these completely in 2020, so I think on the long term none of this really matters, so right now my main focus in getting them wired up in JS not in inline scripts. Lemme know so I can move this along.

Reply:
Just a friendly ping that I'm looking for guidance on how we want to proceed with this change!

Reply:
Hey! Sorry for the long delay; things are starting to stabilize a bit more now :)

I think `data-dialog-href` is a good solution; it communicates a lot of information to any future devs:
- This HTML element has associated javascript (cause otherwise why have a data attr?)
- I can find that javascript by searching for `dialog-href` or `dialogHref`
- If I wanted to create a dialog, this communicates that I'd just have to add `data-dialog-href` attr and an `open` class to make it work up (possible without even looking at the related JS). Reuse.

Putting everything in the JS (without the magic open class) would make this harder to reuse, I fear. if I was editing the HTML and wanted to add a new dialog element, I wouldn't know how to do it.

Reply:
Actually, now that I'm looking at your other PR, is there an aria label we could use?

Reply:
aria-controls would make sense (especially when it's in the DOM)

Reply:
and... done